state/state.go

```diff
@@ -164,15 +164,16 @@ func (state State) MakeBlock(
 // the votes sent by honest processes, i.e., a faulty processes can not arbitrarily increase or decrease the
 // computed value.
 func MedianTime(commit *types.Commit, validators *types.ValidatorSet) time.Time {
-
 	weightedTimes := make([]*tmtime.WeightedTime, len(commit.Precommits))
 	totalVotingPower := int64(0)
 
-	for i, vote := range commit.Precommits {
-		if vote != nil {
-			_, validator := validators.GetByIndex(vote.ValidatorIndex)
-			totalVotingPower += validator.VotingPower
-			weightedTimes[i] = tmtime.NewWeightedTime(vote.Timestamp, validator.VotingPower)
+	for i, commitSig := range commit.Precommits {
+		if commitSig != nil {
+			_, validator := validators.GetByAddress(commitSig.ValidatorAddress)
+			if validator != nil {
```

Comment:
shouldn't we panic at this point?

Reply:
just tried to add a panic. many tests start to fail, hmm

Reply:
maybe the tests aren't set up fully?

Reply:
in which case would a commit happen from a validator that is not in the valSet?

Reply:
> maybe the tests aren't set up fully?

looks like it

> in which case would a commit happen from a validator that is not in the valSet?

that should never happen imo